### PR TITLE
Fix up stpyfits and its tests

### LINKED ISSUE #47: stpyfits uses private classes + methods fits.hdu.image._ImageBaseHDU._summary and fails
I am just trying to create a Debian package from stsci.tools (to use with pyraf), and found that `lib/stsci/tools/stpyfits.py ` accesses methods that are for astropy internal use only. Namely `astropy.io.fits.hdu.image._ImageBaseHDU`, and its method `_summary()`. The latter fails with newer astropy versions (tested with 2.0.2), since it introduced an additional member in the middle, which makes the overwriting fail:

```
    def _summary(self):
        summ = super(_ConstantValueImageBaseHDU, self)._summary()
        return (summ[0], summ[1].replace('ConstantValue', '')) + summ[2:]
```

```
ERROR: Test the info convience function in both the fits and stpyfits
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/stsci.tools-3.4.11/.pybuild/pythonX.Y_2.7/build/stsci/tools/tests/testStpyfits.py", line 31, in testInfoConvienceFunction
    stpyfits.info(self.data('o4sp040b0_raw.fits'), output=False),
  File "/build/stsci.tools-3.4.11/.pybuild/pythonX.Y_2.7/build/stsci/tools/stpyfits.py", line 55, in wrapped_with_stpyfits
    retval = func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/astropy/io/fits/convenience.py", line 704, in info
    ret = f.info(output=output)
  File "/usr/lib/python2.7/dist-packages/astropy/io/fits/hdu/hdulist.py", line 953, in info
    summary = hdu._summary()
  File "/build/stsci.tools-3.4.11/.pybuild/pythonX.Y_2.7/build/stsci/tools/stpyfits.py", line 288, in _summary
    return (summ[0], summ[1].replace('ConstantValue', '')) + summ[2:]
AttributeError: 'int' object has no attribute 'replace'
```

After fixing this, many of the tests fail since they do not take into account the additional tuple member.

### PR DESCRIPTION
**Pull request by pllim**
Fix #47, #48, #49, #50, #52 

c/c @olebole

### CODE REVIEW

File: lib/stsci/tools/stpyfits.py [hunk 1]
```diff
@@ -51,7 +55,8 @@ def wrapped_with_stpyfits(*args, **kwargs):
             # BUG: Forcefully disable lazy loading.
             # Lazy loading breaks ability to initialize ConstantValueHDUs
             # TODO: Investigate the cause upstream (astropy.io.fits)
-            kwargs['lazy_load_hdus'] = False
+            if 'write' not in func.__name__:
+                kwargs['lazy_load_hdus'] = False
```
Comment (pllim) [inline lib/stsci/tools/stpyfits.py]: Note: This is dangerous magic. More patching might be needed in the future.

**Comment by mcara** [comment]
@pllim However, tests are failing 😞

**Comment by pllim** [comment]
Tests passed. @mcara , do you want the honor to merge?

**Comment by mcara** [comment]
@pllim Why me? It is a great honor but feel free to go ahead an merge it.

**Comment by pllim** [comment]
> Why me? 

So `git blame` would blame you for the merge commit? Har har... Or rather I don't usually like to merge my own PRs, but I'll make an exception this time.

**Pull request merged**